pkg/api/v1/clients/memory/resource_client.go

```diff
@@ -162,17 +161,14 @@ func (rc *ResourceClient) Write(resource resources.Resource, opts clients.WriteO
 	}
 
 	// mutate and return clone
-	clone := proto.Clone(resource).(resources.Resource)
+	resource = resources.Clone(resource)
 	// initialize or increment resource version
 	meta.ResourceVersion = newOrIncrementResourceVer(meta.ResourceVersion)
-	clone.SetMetadata(meta)
+	resource.SetMetadata(meta)
 
-	rc.cache.Set(key, clone)
+	rc.cache.Set(key, resource)
 
-	// return another clone!
-	clone = proto.Clone(clone).(resources.Resource)
-
-	return clone, nil
+	return resource, nil
```

Comment:
why not return another clone? this is important so that modfications on the returned object will not changed what's stored inside.

Reply:
it already is cloned on line 164